### PR TITLE
chore(core): removed some internal functions

### PR DESCRIPTION
**Pull request by jdalsem**

### CODE REVIEW

File: engine/classes/Elgg/Database/MetadataTable.php [hunk 1]
```diff
@@ -391,7 +391,23 @@ public function getRowsForGuids(array $guids) {
...
+		foreach ($required as $key) {
+			// check that it exists and is something.
+			if (isset($options[$key]) && $options[$key]) {
+				$found = true;
+				break;
+			}
+		}
+		
+		if (!$found) {
+			// requirements not met
```
Comment (jeabakker) [inline engine/classes/Elgg/Database/MetadataTable.php:410]: why isn't this in a protected function (just like in AnnotationsTable) that way it's easier to test ;)

Comment (jeabakker) [reply engine/classes/Elgg/Database/MetadataTable.php:410]: and ``elgg_is_empty()``

File: engine/classes/Elgg/Database/AnnotationsTable.php [hunk 2]
```diff
@@ -388,6 +388,31 @@ public function enableAll(array $options) {
...
+		$required = [
+			'guid', 'guids',
+			'annotation_owner_guid', 'annotation_owner_guids',
+			'annotation_name', 'annotation_names',
+			'annotation_value', 'annotation_values'
+		];
+		
+		foreach ($required as $key) {
+			// check that it exists and is something.
+			if (isset($options[$key]) && $options[$key]) {
```
Comment (jeabakker) [inline engine/classes/Elgg/Database/AnnotationsTable.php:409]: please use ``elgg_is_empty()``

**Review by jeabakker — CHANGES_REQUESTED** [review]

File: engine/lib/river.php [hunk 3]
```diff
@@ -132,7 +132,25 @@ function elgg_get_river_item_from_id(int $id) {
...
+		'object_guid', 'object_guids',
+		'target_guid', 'target_guids',
+		'annotation_id', 'annotation_ids',
+		'view', 'views',
+	];
+
+	$found = false;
+	foreach ($required as $key) {
+		// check that it exists and is something.
+		if (isset($options[$key]) && $options[$key]) {
```
Comment (jeabakker) [inline engine/lib/river.php:147]: use ``elgg_is_empty()``

**Review by jeabakker — CHANGES_REQUESTED** [review]

File: engine/classes/Elgg/Database/MetadataTable.php (hunk 1 above)
Comment (jdalsem) [reply engine/classes/Elgg/Database/MetadataTable.php:410]: code is just used once.. so not separating the function... elgg_is_empty() added to all the places

**Review by jeabakker — APPROVED** [review]

**Review by jeabakker — APPROVED** [review]

**Pull request closed without merging**